apps/api/bronya.config.ts

```diff
@@ -141,25 +160,99 @@ class MyStack extends Stack {
 }
 
 export async function main() {
+  const id = "peterportal-api-next";
+
   const app = new App();
 
-  const stack = new MyStack(app, "peterportal-api-canary");
+  if (!process.env.NODE_ENV) {
+    throw new Error("NODE_ENV not set.");
+  }
+
+  let stage;
+
+  switch (process.env.NODE_ENV) {
+    case "production":
+      stage = "prod";
+      break;
+    case "staging":
+      if (!process.env.PR_NUM) {
+        throw new Error("NODE_ENV was set to staging, but a PR number was not provided.");
+      }
+      stage = `staging-${process.env.PR_NUM}`;
+      break;
+    case "development":
+      stage = "dev";
+      break;
+    default:
+      throw new Error(
+        "Invalid NODE_ENV specified. Valid values are 'production', 'staging', and 'development'.",
+      );
+  }
+
+  const stack = new ApiStack(app, id, stage);
 
   const api = stack.api;
 
   await api.init();
 
   if (isCdk()) {
+    if (stage === "dev") {
+      throw new Error("Cannot deploy this app in the development environment.");
+    }
     const result = await api.synth();
 
-    result.api.addGatewayResponse;
+    result.api.addGatewayResponse(`${id}-${stage}-5xx`, {
+      type: ResponseType.DEFAULT_5XX,
+      statusCode: "500",
+      templates: {
+        "application/json": JSON.stringify({
+          timestamp: "$context.requestTime",
+          requestId: "$context.requestId",
+          statusCode: 500,
+          error: "Internal Server Error",
+          message: "An unknown error has occurred. Please try again.",
+        }),
+      },
+    });
 
-    result.functions;
+    result.api.addGatewayResponse(`${id}-${stage}-404`, {
+      type: ResponseType.MISSING_AUTHENTICATION_TOKEN,
+      statusCode: "404",
+      templates: {
+        "application/json": JSON.stringify({
+          timestamp: "$context.requestTime",
+          requestId: "$context.requestId",
+          statusCode: 404,
+          error: "Not Found",
+          message: "The requested resource could not be found.",
+        }),
+      },
+    });
+
+    const optionsIntegration = new LambdaIntegration(
+      new AwsLambdaFunction(result.api, `${id}-options-handler`, {
+        code: Code.fromInline(
+          // language=JavaScript
+          'exports.h=async _=>({body:"",headers:{"Access-Control-Allow-Origin":"*","Access-Control-Allow-Headers":"Apollo-Require-Preflight,Content-Type","Access-Control-Allow-Methods":"GET,POST,OPTIONS"},statusCode:204});',
+        ),
+        handler: "index.h",
+        runtime: Runtime.NODEJS_18_X,
+        architecture: Architecture.ARM_64,
+      }),
+    );
+
+    result.api.methods.forEach((x) => {
+      try {
+        x.resource.addMethod("OPTIONS", optionsIntegration);
+      } catch {
+        // no-op
+      }
+    });
   }
 
   return app;
 }
 
 if (isCdk()) {
-  main();
+  main().then();
```

Comment:
Would this work alright too?

```suggestion
  await main()
```

Reply:
Yeah that's fine. I keep forgetting we have top level await now lol